### PR TITLE
Bug fix for not saving explicit name and email settings to local config

### PR DESCRIPTION
**Pull request by fulghum**
Fixes: https://github.com/dolthub/dolt/issues/3546

### CODE REVIEW

File: go/cmd/dolt/commands/init_test.go [hunk 1]
```diff
@@ -96,3 +101,37 @@ func TestInitTwice(t *testing.T) {
...
+	for i := 0; i <= len(test.Args)-2; i = i + 2 {
+		if test.Args[i] == "-"+argKey {
+			found = true
+			expectedValue = test.Args[i+1]
+		}
+	}
+
+	actualValue, err := localConfig.GetString(envKey)
+	if found {
+		if err != nil {
```
Comment (zachmu) [inline go/cmd/dolt/commands/init_test.go:125]: Nit: try not to write any new tests without testify

File: integration-tests/bats/init.bats [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  [[ "$output" =~ "foo" ]] || false
+
+  run dolt config --local --get user.email
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "foo@bar.com" ]] || false
+
+  assert_valid_repository
+}
+
+@test "init: no explicit or implicit configuration for name and email" {
```
Comment (zachmu) [inline integration-tests/bats/init.bats:79]: Seems like we're missing a test where `user.name` and `user.email` aren't set, but are given to `dolt init`?

**Review by zachmu — APPROVED** [review]
LGTM, thanks for fixing!

File: go/cmd/dolt/commands/init_test.go (hunk 1 above)
Comment (fulghum) [reply go/cmd/dolt/commands/init_test.go]: ooooh... good call. I migrated this unit test over to stretchr/testify and it's cleaner/leaner now.

File: integration-tests/bats/init.bats (hunk 2 above)
Comment (fulghum) [reply integration-tests/bats/init.bats:96]: I was covering a similar case where global config was set and tested that explicit config was saved to local config, but you're right that it's worth also covering when no global config is present at all. Added now.

**Pull request merged**